### PR TITLE
Add Documentation Issue form template

### PR DESCRIPTION
**Pull request by crw**
Closes https://github.com/hashicorp/terraform/issues/31552. 

Adds a new issue form type for documentation issues. Format discussed with docs team.

### CODE REVIEW

**Comment by crw** [comment]
You can preview what this will look like here:
https://github.com/crw/terraform/issues/new?assignees=&labels=documentation%2Cnew&template=documentation_issue.yml

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:15]:
```suggestion
        Please only use the [hashicorp/terraform](https://github.com/hashicorp/terraform) `Documentation` issue
        type to report problems with the documentation on [https://www.terraform.io/docs]().
        Only technical writers (not engineers) monitor this issue type. Report Terraform bugs or feature requests 
        with the `Bug report` or `Feature Request` issue types instead to get engineering attention.
```

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:35]:
```suggestion
          Link to the pages relevant to your documentation change request.
```

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:45]:
```suggestion
      description: What problems or suggestions do you have about the documentation?
```

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:60]:
```suggestion
          What documentation changes would fix this issue and where
          you would expect to find them? Are one or more page headings unclear? Do one or more pages
          need additional context, examples, or warnings? Do we need a new page or section dedicated
          to a specific topic?  Your ideas help us understand what you and other users need from our
          documentation and how we can improve the content.
```

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:71]:
```suggestion
        Are there any other open or closed GitHub issues related to the problem or solution you described? If so, list them below. For example:
```

**Review by kmoe — APPROVED** [review]
LGTM pending final review from @judithpatudith.

File: .github/ISSUE_TEMPLATE/documentation_issue.yml
Comment (judithpatudith) [inline .github/ISSUE_TEMPLATE/documentation_issue.yml:52]:
```suggestion
         would you expect to find them? Are one or more page headings unclear? Do one or more pages
```

**Review by judithpatudith — APPROVED** [review]
It's beautiful! Fond one teeny last word swap, but this is my official approval! Thank you :D

**Pull request merged**
